source/language/types.rst

```diff
@@ -373,3 +375,174 @@ variables whose values may only be known at run time.
    let last_three = two[-4:-1];
    // Concatenate two alias in another one
    let both = sliced || last_three;
+
+.. _castingSpecifics:
+
+Casting specifics
+-----------------
+
+The classical types are divided into the 'standard' classical types (bool, int, 
+uint, and float) that exist in languages like C, and the 'special' classical 
+types (bit, angle, duration, and stretch) that do not.
+
+In general, for any cast between standard types that results in loss of 
+precision, if the source value is larger than can be represented in the target 
+type, the exact behavior is implementation specific and must be documented by 
+the vendor.
+
+Allowed casts
+~~~~~~~~~~~~~
+
+.. role:: gbg
+
+.. raw:: html
+
+   <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
+   <script>
+     $(document).ready(function() {
+       $('.gbg').parent().addClass('gbg-parent');
+     });
+   </script>
+   <style>
+      .gbg-parent {text-align:center;background-color:#c6efce;color:#006100;}
+   </style>
+
+.. role:: rbg
+
+.. raw:: html
+
+   <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
+   <script>
+     $(document).ready(function() {
+       $('.rbg').parent().addClass('rbg-parent');
+     });
+   </script>
+   <style>
+      .rbg-parent {text-align:center;background-color:#ffc7ce;color:#9c0006;}
+   </style>
+
+.. role:: center
+   
+.. raw:: html
+
+   <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
+   <script>
+     $(document).ready(function() {
+       $('.center').parent().addClass('center-parent');
+     });
+   </script>
+   <style>
+      .center-parent {text-align: center}
+   </style>
```

Comment:
What is all this for?

Reply:
This was the only way I could find to add colors to the casting tables, which I felt was very important for readability.

Reply:
The three styles this adds are red-background (rbg) green-background (gbg) and centered.